### ISSUE TITLE
Do not call Wire.begin inside the sensor.begin

### ISSUE THREAD

**Issue by dizcza** [adafruit/Adafruit_LIS3MDL#13]
Calling `Wire.begin` inside a sensor initialization is a bad practice and developers should avoid it by all means.

https://github.com/adafruit/Adafruit_LIS3MDL/blob/64e685ba784c9741b5b101ca8ce509a101a8ff1b/Adafruit_LIS3MDL.cpp#L51

**Comment by nomakewan** [comment]
I know this is over a year later, but I wanted to point out that this issue is not actually an issue with this library at all. The `begin()` call you're referencing above is *not* a call to the `TwoWire` class, but rather a call to Adafruit BusIO.

Therefore, this issue would be better suited for [Adafruit BusIO](https://github.com/adafruit/Adafruit_BusIO).

Though once you see how this is structured, and considering the massive amount of user code out in the wild that relies on the existing functionality, I think this will be an uphill battle. Best of luck.

**Comment by dizcza** [comment]
Yeah. This is unfortunate. And I even know what would happen: they would respond to me that you can easily call `Wire.begin()` multiple times without an error. They will forget to mention, however, that in this case, you would see a red error message in esp32 native Wire library saying the I2C periphery has already been initialized. Who cares about errors in logs, right.
